src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md

```diff
@@ -0,0 +1,325 @@
+        </div>
+    </div>
+</div>
+```
+
+{% endcollapsible %}
+
+## Step 5: Add an Observer {#add-observer}
+
+Create an Observer file to save the custom field data to the order. For the Observer file an `event.xml` file is required to call the observer for a particular event. For this example, the `checkout_onepage_controller_success_action` event is used.
+
+`app/code/Learning/CustomField/etc/frontend/events.xml`
+
+```xml
+<?xml version="1.0"?>
+
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Event/etc/events.xsd">
+    <event name="checkout_onepage_controller_success_action">
+        <observer name="paymentfields_paymentfields_observer_frontend_sales_orderpaymentsavebefore" instance="Learning\CustomField\Observer\Frontend\Sales\OrderPaymentSaveBefore" />
+    </event>
+</config>
+```
+
+`app/code/Learning/CustomField/Observer/Frontend/Sales/OrderPaymentSaveBefore.php`
+
+{% collapsible Show code %}
+
+```php?start_inline=1
+<?php
+
+namespace Learning\CustomField\Observer\Frontend\Sales;
+
+use Magento\Framework\Event\Observer as EventObserver;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\OfflinePayments\Model\Purchaseorder;
+use Magento\Framework\App\Request\DataPersistorInterface;
+
+class OrderPaymentSaveBefore implements \Magento\Framework\Event\ObserverInterface
+{
+    protected $order;
+    protected $logger;
+    protected $_serialize;
+    protected $quoteRepository;
+
+    public function __construct(
+        \Magento\Sales\Api\Data\OrderInterface $order,
+        \Magento\Quote\Api\CartRepositoryInterface $quoteRepository,
+        \Psr\Log\LoggerInterface $logger,
+        \Magento\Framework\Serialize\Serializer\Serialize $serialize
+    ) {
+        $this->order = $order;
+        $this->quoteRepository = $quoteRepository;
+        $this->logger = $logger;
+        $this->_serialize = $serialize;
+    }
+    /**
+     * Execute observer
+     *
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return void
+     */
+    public function execute(\Magento\Framework\Event\Observer $observer)
+    {
+        $orderids = $observer->getEvent()->getOrderIds();
+        if(!$orderids){
+            foreach ($orderids as $orderid) {
+                $order = $this->_order->load($orderid);
+                $method = $order->getPayment()->getMethod();
+                if($method == 'purchaseorder') {
+                    $quote_id = $order->getQuoteId();
+                    $quote = $this->quoteRepository->get($quote_id);
+                    $paymentQuote = $quote->getPayment();
+                    $paymentOrder = $order->getPayment();
+                    $paymentOrder->setData('paymentpocomment',$paymentQuote->getPaymentpocomment());
+                    $paymentOrder->save();
+                }
+            }
+        }
+    }
+}
+```
+
+{% endcollapsible %}
+
+## Step 6: Verify that the module works {#verify-implementation}
+
```

Comment:
@keharper 
Yes, since we are creating a new module, we need to run the following commands:

```
bin/magento setup:upgrade
bin/magento setup:di:compile
bin/magento setup:static-content:deploy
```

Do you want me to add these commands as well before the verify step?

Reply:
Yes

Reply:
Added...